src/improvements/doc/TASK_CREATION_GUIDE.md

```diff
@@ -0,0 +1,196 @@
+├── oeth/          # Optimism Ethereum tasks
+├── opsep/         # Optimism Sepolia tasks
+└── common/        # Shared resources and documentation
+```
+
+### Network-Specific Organization
+
+1. Ethereum Mainnet (tasks/eth/)
+   - Production tasks
+   - Strict security requirements
+   - Full testing required
+   - Example: eth/001-security-council-phase-0/
+
+2. Sepolia Testnet (tasks/sep/)
+   - Testing and validation
+   - Development tasks
+   - Example: sep/001-op-extended-pause/
+
+3. Optimism Networks
+   - tasks/oeth/: Optimism Ethereum tasks
+   - tasks/opsep/: Optimism Sepolia tasks
+   - Network-specific configurations required
+
+### Task Naming Convention
+
+Tasks follow a numbered sequence within each network directory:
+- Format: `[network]/[###]-[descriptive-name]`
+- Example: `eth/001-security-council-phase-0`
+- Numbers help track task order and dependencies
+- Descriptive names should be clear and concise
+
+## Creating a New Task
+
+### 1. Use the Task Scaffolding Tool
+
+```bash
+cd src/improvements/
+just new task
+```
+
+The tool will:
+- Create a new numbered directory
+- Generate README.md template
+- Create config.toml template
+- Set up basic structure
+
+### 2. Configure the Task
+
+Edit config.toml to specify:
+
+1. Template Selection
+```toml
+templateName = "GasConfigTemplate"
+```
+
+2. L2 Chain Configuration
+```toml
+l2chains = [
+    {name = "Orderly", chainId = 291},
+    {name = "Metal", chainId = 1750}
+]
+```
+
+3. Template-Specific Parameters
+```toml
+[gasConfigs]
+gasLimits = [
+    {chainId = 291, gasLimit = 100000000},
+    {chainId = 1750, gasLimit = 100000000}
+]
+```
+
+### 3. Document the Task
+
+Update README.md with:
+- Task description
+- Requirements
+- Expected outcomes
+- Testing steps
+- Validation criteria
+- Status updates
+
+## Testing and Validation
+
+1. Local Testing
+```bash
+forge script <template-path> --sig "run(string)" <config-file-path> --rpc-url devnet -vvv
+```
+
+2. Testnet Validation
+- Deploy to Sepolia first
+- Verify state changes
+- Check validation results
+
+3. Production Deployment
+- Full security review
+- Comprehensive testing
+- Stakeholder approval
+
+## Common Task Types
+
+1. Gas Configuration
+```toml
+templateName = "GasConfigTemplate"
+[gasConfigs]
+gasLimits = [...]
+```
+
+2. Dispute Game Upgrades
+```toml
+templateName = "DisputeGameUpgradeTemplate"
+implementations = [...]
+```
+
+3. Game Type Configuration
+```toml
+templateName = "SetGameTypeTemplate"
+respectedGameTypes = [...]
+```
+
+## Best Practices
+
+1. Task Organization
+   - Use clear, descriptive names
+   - Follow network-specific conventions
+   - Maintain proper documentation
+
+2. Configuration
+   - Document all parameters
+   - Use consistent formatting
+   - Include validation checks
+
+3. Documentation
+   - Keep README.md updated
+   - Document any special requirements
+   - Include validation steps
+   - Track task status
+
+## Troubleshooting
+
+1. Configuration Issues
```

Comment:
Similarly, can we remove this? Once we've got a fully production ready end to end flow working we can decide how much troubleshooting guides to add.